# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmGenerator.cpp
# Commit: 1a3bc285395f
# Full Hash: 1a3bc285395ffcc5d9281dd45d4f6c2bb918bf43
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-03-04 04:58:00
# Regressor Bug: 1861078
# File Overlap Count: 1
# Description:
#   Bug 1861078 - wasm: make metadata offsets relative to CodeBlock->codeBase. r=rhunt
#   
#   Metadata used offsets to the beginning of code segment. Change that to use
#   offsets relative to the beginning of CodeBlock. This allows us to avoid offseting
#   metadata later on and prepare the work for simplification of stackmaps.
# ==============================================================================

diff -r 66d70b730ea6 -r 1a3bc285395f js/src/wasm/WasmGenerator.cpp
--- a/js/src/wasm/WasmGenerator.cpp	Mon Mar 03 16:15:17 2025 +0000
+++ b/js/src/wasm/WasmGenerator.cpp	Mon Mar 03 16:37:47 2025 +0000
@@ -927,12 +927,6 @@
     }
     codeBlock_->codeBase = codeStart;
     codeBlock_->codeLength = codeLength;
-
-    // All metadata in code block is relative to the start of the code segment
-    // we were placed in, so we must adjust offsets for where we were
-    // allocated.
-    uint32_t codeBlockOffset = codeStart - codeBlock_->segment->base();
-    codeBlock_->offsetMetadataBy(codeBlockOffset);
   } else {
     // Create a new CodeSegment for the code and use that.
     codeBlock_->segment = CodeSegment::createFromMasm(